AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -13,17 +13,27 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
     
     /// :nodoc:
     public func paymentAuthorizationViewControllerDidFinish(_ controller: PKPaymentAuthorizationViewController) {
-        dismiss { [weak self] in
-            guard let self = self, self.success == false else { return }
+        switch state {
+        case .initiated:
             self.delegate?.didFail(with: ComponentError.cancelled, from: self)
+        case .paymentInProgress:
+            break // viewController should be dissmissed when API call is proccessed
```

Comment:
`paymentAuthorizationViewControllerDidFinish` can be called when user wants to cancel, right? does this mean the shopper cannot cancel while the payment is in progress? did you try this scenario?

Reply:
> shopper cannot cancel while the payment is in progress?

This is correct. It is possible for about half a second after starting a payment (in other words - if the shopper is determined to do so), but basically - after one passes the authorization step (Face ID, Passcode, etc) Apple Pay will remain on a screen till of API call is conducted.